File: be-web/app/api/routes/notifications.py
from fastapi import APIRouter, Depends

from app.domain.models.user import User
from app.services.notification_service import NotificationService
from app.schemas.notification import NotificationResponse, NotificationListResponse
from app.api.dependencies import get_current_user, get_notification_service

router = APIRouter(prefix="/notifications", tags=["Notifications"])


@router.get("/", response_model=NotificationListResponse)
def list_notifications(
    skip: int = 0,
    limit: int = 50,
    current_user: User = Depends(get_current_user),
    service: NotificationService = Depends(get_notification_service),
):
    return service.get_user_notifications(current_user.id, skip, limit)


@router.patch("/{notification_id}/read", response_model=NotificationResponse)
def mark_notification_read(
    notification_id: int,
    current_user: User = Depends(get_current_user),
    service: NotificationService = Depends(get_notification_service),
):
    return service.mark_as_read(current_user.id, notification_id)


@router.post("/read-all")
def mark_all_read(
    current_user: User = Depends(get_current_user),
    service: NotificationService = Depends(get_notification_service),
):
    return service.mark_all_read(current_user.id)

File: fe-web/src/api/notifications.js
/**
 * Notifications API — list, mark read.
 */
import api from './client';

export const notificationsApi = {
  /** List current user's notifications */
  list(skip = 0, limit = 50) {
    return api.get(`/notifications/?skip=${skip}&limit=${limit}`);
  },

  /** Mark a single notification as read */
  markRead(notificationId) {
    return api.patch(`/notifications/${notificationId}/read`);
  },

  /** Mark all notifications as read */
  markAllRead() {
    return api.post('/notifications/read-all');
  },
};

export default notificationsApi;
